### PR TITLE
chore(performance): Disable jemallocator background threads on OSX

### PR DESCRIPTION
**Pull request by jszwedko**
These appear to be unsupported:
https://github.com/jemalloc/jemalloc/issues/843

Build failure: https://github.com/timberio/vector/runs/1428614738

Signed-off-by: Jesse Szwedko <jesse@szwedko.me>

### CODE REVIEW

**Review by jamtur01 — APPROVED** [review]
LGTM

**Review by lukesteensen — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -4,12 +4,14 @@
 # Begin OS detection
 ifeq ($(OS),Windows_NT) # is Windows_NT on XP, 2000, 7, Vista, 10...
     export OPERATING_SYSTEM := Windows
-	export RUST_TARGET ?= "x86_64-unknown-windows-msvc"
```
Comment (jszwedko) [inline Makefile:7]: @Hoverbear do you know why these `RUST_TARGET` vars were here? I couldn't find them referenced anywhere in here or by `cross`, `rustc`, or `cargo`.

Comment (Hoverbear) [reply Makefile:7]: Hmmm it think they are unused now. It may have been related to support for both gnu and msvc on Windows.

**Comment by jszwedko** [comment]
OSX test failure appears to be a flake, opened https://github.com/timberio/vector/issues/5154

**Pull request merged**